app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -333,7 +333,30 @@ private void runInitSequence(Boolean loadCalibrationData) throws IOException {
                             }
                         }
 
-                        // todo : port remaining function
+                        ArrayUtils.reverse(fitVals);
+                        PolynomialFunction fitFunction = new PolynomialFunction(fitVals);
+                        double[] yData = new double[DACX.length];
+                        for (int j = 0; j < DACX.length; j++) {
+                            yData[j] = fitFunction.value(DACX[j]) - (OFF.get(j) * slope + intercept);
+                        }
+                        int LOOKBEHIND = 100;
+                        int LOOKAHEAD = 100;
```

Comment:
Hi @viveksb007 ! If both these values are having the same value, can we assign a single variable instead of two and use it in the code?